File: ApiChessWebApp/Piece.cs

using ApiChessWebApp.Models;

namespace ApiChessWebApp
{
    public enum Colors
    {
        Black,
        White
    }
    public class Piece
    {
        public bool IsMovingFirstTime { get; set; } = true;//because first time pawn can move 2 steps
        private Cordinates PieceCordinates { get; set; }
        public Colors? ColorCode{ get; set; }
        public string? Color { get; set; }
        public bool IsWhite{ get; set; }
        public virtual string TypeOfPiece { get; set; }
        public virtual string HtmlCode { get; set; } = "";
        public Piece(){}
        public Piece(int x, int y){this.PieceCordinates = new Cordinates(x, y);}
        public Piece(int x, int y, char z){this.PieceCordinates = new Cordinates( x,  y, z);}
        public Piece(int x, int y, char z, Colors c)
        {
            this.PieceCordinates = new Cordinates(x, y, z); 
            this.ColorCode = c;
            this.IsWhite = ColorCode == Colors.White;
            this.Color = c.ToString();
        }
        public virtual bool CanMove(Spot from, Spot to, List<List<Spot>> boardSpotsState)
        {
            return true;
        }
    }
}

File: UiChessWebApp/src/app/Helper/classes/Piece.ts
import { Colors } from "../constants/Color";
import { PieceTypes } from "../constants/PieceTypes";
import { Cordinates } from "./Cordinates";

export class Piece{
 isMovingFirstTime: boolean;
    colorCode: number;
    color: string;
    isWhite: boolean;
    typeOfPiece: string;
    htmlCode: string;

    constructor(
        isMovingFirstTime: boolean,
        colorCode: number,
        color: string,
        isWhite: boolean,
        typeOfPiece: string,
        htmlCode: string
    ) {
        this.isMovingFirstTime = isMovingFirstTime;
        this.colorCode = colorCode;
        this.color = color;
        this.isWhite = isWhite;
        this.typeOfPiece = typeOfPiece;
        this.htmlCode = htmlCode;
    }


  // PieceCordinates:Cordinates;
//   Color:Colors;
// Type:PieceTypes;

// /**
//  *
//  */
// constructor(cordinates:Cordinates , color:Colors,types:PieceTypes) {
//    // super();
//     this.PieceCordinates=cordinates;
//     this.Color=color;
//     this.Type=types;
// }
}